routing/heap.go

```diff
@@ -51,9 +51,9 @@ type distanceHeap struct {
 
 // newDistanceHeap initializes a new distance heap. This is required because
 // we must initialize the pubkeyIndices map for path-finding optimizations.
-func newDistanceHeap() distanceHeap {
+func newDistanceHeap(numNodes int) distanceHeap {
 	distHeap := distanceHeap{
-		pubkeyIndices: make(map[route.Vertex]int),
+		pubkeyIndices: make(map[route.Vertex]int, numNodes),
```

Comment:
Do you also want to set the capacity of the `distanceHeap.nodes` slice?

Reply:
`numNodes` is really an upper bound for the size of the heap. It contains only the active set, the frontier of what path finding is exploring. Worst case the frontier can be all nodes (one center node with all other nodes connected to it), but it will be less typically. I don't think it is problem and also wouldn't know how to get a better estimated of the required size for the heap. But maybe this is something that can be explained (if the reasoning is correct) in a comment.